### PR TITLE
DOC: mailing list to forum

### LINKED ISSUE #20603: DOC: document switch from mailing list to discourse
Putting a milestone on this so that it isn't forgotten.

### PR DESCRIPTION
**Pull request by tupui**
Closes #20603

Putting this for 1.15, but would be good if that ended up in 1.14. It's ok I guess.

cc @tylerjereddy as you can.

### CODE REVIEW

File: doc/source/dev/governance.rst [hunk 1]
```diff
@@ -189,7 +189,7 @@ responsible for:
   feedback/suggestions on the mailing list.
```
Comment (lucascolley) [inline doc/source/dev/governance.rst:189]: one more here

**Review by lucascolley — COMMENTED** [review]
I can still see one link on line 20 of `doc/source/index.rst` that should probably change to the forum. You can also link to the user part of the Discourse forum there.

It probably makes sense to keep a link to the mailing list archive somewhere accessible - maybe in the README?

**Comment by tupui** [comment]
I am not sure about the ref to the mailing list, we have that on the main website. But feel free to push a suggestion on this branch.

**Review by lucascolley — APPROVED** [review]
thanks Pamphile, LGTM! We can save adding a link to the User forum for if/when it starts to see more use.

**Pull request merged**